app/src/main/java/com/github/se/travelpouch/model/gps/GPSViewModel.kt

```diff
@@ -0,0 +1,70 @@
+package com.github.se.travelpouch.model.gps
+
+import android.content.Context
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import com.google.android.gms.location.LocationServices
+import com.google.android.gms.maps.model.LatLng
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.launch
+
+class GPSViewModel(private val gpsRepository: GPSRepository) : ViewModel() {
+
+  // Holds the real-time continuous location updates as a StateFlow
+  private val _realTimeLocation = MutableStateFlow<LatLng?>(null)
+  val realTimeLocation: StateFlow<LatLng?>
+    get() = _realTimeLocation
+
+  // Holds the single location update as a StateFlow
+  private val _singleLocationUpdate = MutableStateFlow<LatLng?>(null)
+  val singleLocationUpdate: StateFlow<LatLng?>
+    get() = _singleLocationUpdate
+
+  fun startRealTimeLocationUpdates() {
```

Comment:
It could be nice to have a small description of the function as we have in the rest of the project

Reply:
yes !